packages/forklift-console-plugin/src/modules/Plans/views/create/PlanCreatePage.tsx

```diff
@@ -59,65 +75,68 @@ export const PlanCreatePage: React.FC<{ namespace: string }> = ({ namespace }) =
   });
   useSaveEffect(state, dispatch);
 
-  const steps = [
-    {
-      id: 'step-1',
-      name: 'Select source provider',
-      component: (
-        <SelectSourceProvider
-          projectName={projectName}
-          filterState={filterState}
-          filterDispatch={filterDispatch}
-          dispatch={dispatch}
-          state={state}
-          providers={providers}
-          selectedProvider={selectedProvider}
-        />
-      ),
-      enableNext: filterState?.selectedVMs?.length > 0,
-    },
-    {
-      id: 'step-2',
-      name: 'Create migration plan',
-      component: (
-        <ProvidersCreateVmMigrationPage
-          state={state}
-          dispatch={dispatch}
-          emptyContext={emptyContext}
-        />
-      ),
-      enableNext:
-        !emptyContext &&
-        !(
-          !!state?.flow?.apiError ||
-          Object.values(state?.validation || []).some((validation) => validation === 'error') ||
-          state?.validation?.planName === 'default'
-        ),
-      canJumpTo: filterState?.selectedVMs?.length > 0,
-      nextButtonText: 'Create migration plan',
-    },
-  ];
+  const anyValidationError = Object.values(state?.validation || []).some(
+    (validation) => validation === 'error',
+  );
+
+  const planNameValidationDefault = state?.validation?.planName === 'default';
```

Comment:
consider useMemo

Reply:
The state is fetched using the `useFetchEffects` and I can't really calculate all the dependencies, so there is no real advantage in using `useMemo` here.

Reply:
what do u mean by I cant calculate all dependences? 
there is only 1 dep which is state , u have many other values in this file that will cause to rerender, that arent related to the state

Reply:
The whole `State` changed after almost every change so it's not effective (checked).  I added `useMemo` and changed the dep to use the specific `state?.validation?.planName`  here...